src/compiler-src.md

```diff
@@ -16,22 +19,11 @@ The repository consists of three main directories:
 
 - `compiler/` contains the source code for `rustc`. It consists of many crates
   that together make up the compiler.
-
 - `library/` contains the standard libraries (`core`, `alloc`, `std`,
   `proc_macro`, `test`), as well as the Rust runtime (`backtrace`, `rtstartup`,
   `lang_start`).
-
 - `src/` contains the source code for rustdoc, clippy, cargo, the build system,
-  language docs, etc.
-
-## Standard library
```

Comment:
My feeling is that this section isn't needed, but I'm happy to keep it if people want.

Reply:
I'd rather keep it.

Reply:
Moved this below, the `##` headings are in the same order as the list.

Reply:
👍 Moved this below, so the ## headings are in the same order as the list.